### PR TITLE
fix(tree): fix nested node is not working

### LINKED ISSUE #10437: Nested tree not load async nodes with AOT
#### Bug, feature request, or proposal:
Bug
#### What is the expected behavior?
Building the app with aot and click item icon, it will load children and update view with children of the node.

#### What is the current behavior?
Building the app with aot and click item icon, it will load children and not show children of the node.

#### What are the steps to reproduce?
- Clone the repo: https://github.com/DocLM/nested-tree-example
- Install dependencies
- Run with `ng serve --aot`
- Open the page and click on the `>` icon.

Live example (here it work): https://stackblitz.com/edit/angular-olseyg

#### What is the use-case or motivation for changing an existing behavior?
It work without aot and not work with aot

#### Which versions of Angular, Material, OS, TypeScript, browsers are affected?
Browsers: Chrome 65 on MacOS 10.13.3
Material: 6.0.0-beta.4
Angular: 6.0.0-beta.7 (also tested with 5)
Typescript: ~2.5.3

### PR DESCRIPTION
**Pull request by tinayuangao**
Fixes #10437

### CODE REVIEW

File: src/lib/tree/node.ts [hunk 1]
```diff
@@ -101,4 +101,12 @@ export class MatNestedTreeNode<T> extends _MatNestedTreeNodeMixinBase<T>
...
     this.tabIndex = Number(tabIndex) || 0;
   }
+
+  ngAfterContentInit() {
+    super.ngAfterContentInit();
+  }
+
+  ngOnDestroy() {
+    super.ngOnDestroy();
+  }
```
Comment (jelbourn) [inline src/lib/tree/node.ts]: Is there a test that would have caught this?

Comment (tinayuangao) [reply src/lib/tree/node.ts]: It only happen in aot or prod mode. We have test cover nested tree, but I'm not sure how can we run tests in aot mode.

Comment (jelbourn) [reply src/lib/tree/node.ts]: Hmm... something seems odd about this. Neither of those modes should affect the class construction, and the child classes should always inherit its base class's methods. Do you have a reproduction w/ aot?

Comment (tinayuangao) [reply src/lib/tree/node.ts]: Done. Thanks for review

**Pull request merged**